enumerable.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    i = 0
+    array2 = []
+    while i < size
+      array2 << self[i] if yield(self[i])
+      i += 1
+    end
+    array2
+  end
+
+  def my_all?
+    i = 0
+    while i < size
+      return false unless yield(self[i])
+
+      i += 1
+    end
+    true
+  end
+
+  def my_any?
+    i = 0
+    while i < size
+      return true if yield(self[i])
+
+      i += 1
+    end
+    false
+  end
+
+  def my_none?
+    i = 0
+    while i < size
+      return false if yield(self[i])
+
+      i += 1
+    end
+    true
+  end
```

Comment:
**[REQUIRED]**
*When no block or argument is given returns true only if none of the collection members is true.
```
Failure/Error: expect(false_array.my_none?).to be true
      
      LocalJumpError:
        no block given (yield)
```
* When a class is passed as an argument returns true if none of the collection is a member of such class.
```
Failure/Error: expect(array.my_none?(String)).to be true
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)Failure/Error: expect(array.my_none?(String)).to be true
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)
```
* When a Regex is passed as an argument returns true only if none of the collection matches the Regex.
```
Failure/Error: expect(words.my_none?(/z/)).to be words.none?(/z/)
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)
```
* When a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern.
```
Failure/Error: expect(words.my_none?(5)).to be words.none?(5)
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)
```